pystac/utils.py

```diff
@@ -10,11 +11,27 @@
 
 from pystac.errors import RequiredPropertyMissing
 
+HREF = Union[str, os.PathLike]
+
 # Allow for modifying the path library for testability
 # (i.e. testing Windows path manipulation on non-Windows systems)
 _pathlib = os.path
```

Comment:
Since we're working with windows stuff in this PR, can we remove this alias and close https://github.com/stac-utils/pystac/issues/215 as well?

Reply:
We can. I wasn't keen on wrapping another issue into this PR, but it's feeling pretty solid now, so I'll update to close #215.

Reply:
Done.